proxy/logging/LogObject.cc

```diff
@@ -630,8 +637,10 @@ LogObject::log(LogAccess *lad, const char *text_entry)
   } else if (lad) {
     bytes_used = m_format->m_field_list.marshal(lad, &(*buffer)[offset]);
     ink_assert(bytes_needed >= bytes_used);
-  } else if (text_entry) {
-    ink_strlcpy(&(*buffer)[offset], text_entry, bytes_needed);
+  } else if (!text_entry.empty()) {
+    char *dst = &(*buffer)[offset];
+    memcpy(dst, text_entry.data(), text_entry.size());
+    memset(dst + text_entry.size(), 0, bytes_needed - text_entry.size());
```

Comment:
Should just need to do `buffer[bytes_needed] = '\0';` instead of calling memset for setting NULL.

Reply:
I dislike leaving uninitialized memory in the log buffer. It's at most 6 more bytes, I think it's worth it to have clean buffers.

Reply:
Although, actually, you'd need `buffer[text_entry.size()] = 0;`. Setting the byte at `bytes_needed` could leave uninitialized memory between the end of the string and the null terminator.